Allow PHP 8.2 for Phar usage in composer.json

I am using psalm as a phar via phive. Executing it results in the following error message. 

```
Box Requirements Checker
========================
> Using PHP 8.2.0RC6
> PHP is using the following php.ini file:
  WARNING: No configuration file (php.ini) used by PHP!
> Checking Box requirements:
  E...........
                                                                                
 [ERROR] Your system is not ready to run the application.                       
                                                                                
Fix the following mandatory requirements:
=========================================
 * The application requires the version "^7.4 || ~8.0.0 || ~8.1.0" or greater.
```

I am not sure but I assume that the require php value in composer.json should be updated to allow PHP 8.2 and thereby allow the execution in PHP 8.2. The composer.json contains exactly the same statement as shown in the last line of the error message: 
```json
[...]
  "require": {
        "php": "^7.4 || ~8.0.0 || ~8.1.0",
[...]
```

--- a/src/Psalm/Internal/Type/TypeParser.php
+++ b/src/Psalm/Internal/Type/TypeParser.php
@@ -90,7 +90,6 @@ use function is_numeric;
 use function preg_match;
 use function preg_replace;
 use function reset;
-use function str_starts_with;
 use function stripslashes;
 use function strlen;
 use function strpos;
@@ -1472,7 +1471,7 @@ class TypeParser
             return new TObjectWithProperties($properties, [], [], $from_docblock);
         }
 
-        $callable = str_starts_with($type, 'callable-');
+        $callable = strpos($type, 'callable-') === 0;
         $class = TKeyedArray::class;
         if ($callable) {
             $class = TCallableKeyedArray::class;

